### PR TITLE
feat(v-select): additional slots before/after options list

### LINKED ISSUE #4061: [Feature Request] v-select list slot
### Problem to solve
now we don't have opportunity for customize VSelectList, only items are customizable via slots
for example, at VSelectList footer we can show info about available values(current count vs total)

### Proposed solution
add slots after/append, before/prepend for options list

### PR DESCRIPTION
**Pull request by fearoff999**
## Description
Adds two new slots(before-list, after-list) to v-select options list

## Motivation and Context
Need to be able to customize these parts of v-select(autocomplete) menu, and think that the best option to make it real is create a pull request:)
Resolves #4061

## How Has This Been Tested?
Visual

## Markup:

<details>

```vue
<template>
  <v-select
    :items="[{title:'test1',id:1},{title:'test2',id:2}]"
    item-text="title"
    item-value="id"
    autocomplete
    hide-details
    hide-selected
    clearable
    tags
  >
    <template slot="prepend-item">
      some content before list (i.e. 'load more' or 'select all' btn)
    </template>
    <template slot="item" slot-scope="data">
      <v-list-tile-content>
        <v-list-tile-title v-html="data.item['title']" />
      </v-list-tile-content>
    </template>
    <template slot="append-item">
      some content after list (i.e 'scroll to top' btn or info msg such as 'records #k# of #n#')
    </template>
  </v-select>
</template>
```
</details>

## Types of changes
new feature

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by johnleider** [comment]
Thank you for your Pull Request and interest in improving Vuetify. As `v-autocomplete` is part of a series of components that we call _trap components_, I'm curious as to the other @vuetifyjs/dev-team 's thoughts on this functionality. It feels oddly specific and something that I feel could be solved in user land without the need to add this code.

File: src/components/VSelect/VSelect.js [hunk 1]
```diff
@@ -400,18 +400,30 @@ export default {
...
         return this.genListWithSlot()
       } else {
         return this.staticList
       }
     },
     genListWithSlot () {
-      return this.$createElement(VSelectList, this.listData, [
-        this.$createElement('template', {
+      const slots = []
+      if (this.$slots['no-data']) {
```
Comment (jacekkarczmarczyk) [inline src/components/VSelect/VSelect.js]: how about `const slots = ['before-list', 'no-data', 'after-list'].map(slotName => this.$createElement('template', ...)`?

**Comment by jacekkarczmarczyk** [comment]
Thank you, looks much cleaner, you need also check if the slot exists (like you did in your previous code, I just didn't include it in my snippet)

**Comment by fearoff999** [comment]
@jacekkarczmarczyk done:)

**Comment by KaelWD** [comment]
> something that I feel could be solved in user land

I don't think this *can* be solved in user land. It will also allow for even more flexibility, such as somewhere to add "select all" or "load more" buttons.

**Comment by johnleider** [comment]
Fair enough @KaelWD , those are 2 perfectly acceptable use-cases. Let's get the conflicts resolved and merge this in.

**Comment by jacekkarczmarczyk** [comment]
We're using "prepend" and "append" across the framework instead of "before"/"after", maybe slot names should be "prepend-item" and "append-item"?

**Comment by fearoff999** [comment]
@jacekkarczmarczyk done

**Comment by johnleider** [comment]
I tested this using this markup:

``` vue
<template>
  <boilerplate>
    <v-combobox
        v-model="selected"
        :items="[{title:'test1',id:1},{title:'test2',id:2}]"
        item-text="title"
        item-value="id"
        clearable
        hide-details
        hide-selected
        multiple
    >
        <template slot="before">
            some content before list
        </template>
        <template slot="item" slot-scope="data">
            <v-list-tile-content>
                <v-list-tile-title v-html="data.item['title']" />
            </v-list-tile-content>
        </template>
        <template slot="after">
            some content after list
        </template>
    </v-combobox>
  </boilerplate>
</template>
<script>
export default {
  name: 'AutoCompleteWithSlots',
  data () {
      return {
          selected: [],
      }
   }
}
</script>
```

Was not able to get it to work.

**Comment by fearoff999** [comment]
@johnleider forgot to update markup after renaming slots( @jacekkarczmarczyk suggestion ). Now conflicts solved and markup updated

**Pull request merged**